### PR TITLE
topicList: Add an OfflineNotice component.

### PR DESCRIPTION
**Pull request by codebu5ter**
This adds an OfflineNotice component to the topicList to let the user know that they are offline, instead of just showing the LoadingIndicator spinning endlessly.

screenshot_20180414-110127

EDIT: Updated the screenshot.

### CODE REVIEW

**Comment by codebu5ter** [comment]
The padding around the OfflineNotice is due to the padding defined in the TopicListScreen. Shall I remove that padding?

**Comment by borisyankov** [comment]
This approach is not very scalable. We will need to add this to every component manually.

**Comment by codebu5ter** [comment]
@borisyankov I've now reworked my commit a bit. I've now added a showOfflineNotice prop to the Screen component, so that any screen requiring to show the OfflineNotice can just pass the showOfflineNotice prop to do so. This will scale much better and will require minimal effort to implement.

File: src/common/Screen.js [hunk 1]
```diff
@@ -36,6 +36,7 @@ type Props = {
   search?: boolean,
   title?: LocalizableText,
   searchBarOnChange?: (text: string) => void,
+  showOfflineNotice?: boolean,
```
Comment (jainkuniya) [inline src/common/Screen.js]: (clean up: just add it above method)

File: src/common/Screen.js [hunk 2]
```diff
@@ -60,6 +61,7 @@ class Screen extends PureComponent<Props> {
       search,
       searchBarOnChange,
       title,
+      showOfflineNotice,
```
Comment (jainkuniya) [inline src/common/Screen.js]: (clean up: add in alphabetically)

File: src/common/Screen.js (hunk 1 above)
Comment (codebu5ter) [reply src/common/Screen.js]: Done. 👍

File: src/common/Screen.js (hunk 2 above)
Comment (codebu5ter) [reply src/common/Screen.js]: Changes done, as requested. 👍

**Comment by borisyankov** [comment]
@codebu5ter This PR has stalled a bit.
When reviewing it with @gnprice last week we thought it might be good to merge.

After a more careful examination though I think otherwise.

First thing, the code now is clear, concise and correct. Thanks for making it so.

My recommendation not to implement the offline notice this way is because we use it only once in the `TopicListScreen` and it doesn't look like we will need it in another `Screen`. Thus, just including the `<OfflineNotice />` manually in the `TopicListScreen` just above the `<TopicList />` will be simplest.

We might not even want to do this because we are considering moving the topics to another place in the UI (but no final decision made yet)

**Comment by borisyankov** [comment]
@gnprice I think we can close this PR now.

The alternative suggested by me, to include the `<OfflineNotice />` in `TopicListScreen` is a better and shorter solution, but an even better solution, one that we discussed and should do relatively soon, is to correctly indicate a failed loading of the topic data, which will correlate with no connection.

**Comment by borisyankov** [comment]
I decided to add the `<OfflineNotice />` anyway and after testing, it seems Robert did add this correctly during the summer! :)

https://github.com/zulip/zulip-mobile/blob/master/src/common/Screen.js#L111

This can now be closed as implemented!

**Comment by gnprice** [comment]
:smile: Great!

**Pull request closed without merging**